tests/data_array_test.py

```diff
@@ -378,3 +378,58 @@ def test_zeros_like_deep_copy_masks():
     a.masks['m'][0] = False
     assert sc.identical(b.coords['x'][0], c)
     assert sc.identical(b.masks['m'][0], sc.scalar(True))
+
+
+def make_drop_example_dataarray(coord_to_be_dropped: str = None,
+                                mask_to_be_dropped: str = None,
+                                attr_to_be_dropped: str = None):
+    data = sc.array(dims=['x', 'y'], values=np.random.rand(4, 3))
+
+    coord0 = sc.linspace('x', start=0.2, stop=1.61, num=4)
+    coord1 = sc.linspace('y', start=1, stop=4, num=3)
+    coords = {'coord0': coord0, 'coord1': coord1}
+    dropped_coords = coords.copy()
+    if coord_to_be_dropped is not None:
+        del dropped_coords[coord_to_be_dropped]
+
+    mask0 = sc.array(dims=['x'], values=[False, True, True, False], unit='')
+    mask1 = sc.array(dims=['y'], values=[1, 2, 4])
+    masks = {'mask0': mask0, 'mask1': mask1}
+    dropped_masks = masks.copy()
+    if mask_to_be_dropped is not None:
+        del dropped_masks[mask_to_be_dropped]
+
+    attr0 = sc.scalar('attribute 0')
+    attr1 = sc.linspace('x', start=0.2, stop=1.61, num=4)
+    attrs = {'attr0': attr0, 'attr1': attr1}
+    dropped_attrs = attrs.copy()
+    if attr_to_be_dropped is not None:
+        del dropped_attrs[attr_to_be_dropped]
+
+    original_da = sc.DataArray(data, coords=coords, attrs=attrs, masks=masks)
+    expected_da = sc.DataArray(data,
+                               coords=dropped_coords,
+                               masks=dropped_masks,
+                               attrs=dropped_attrs)
+    return original_da, expected_da
+
+
+def test_drop_coords():
+    coord_to_be_dropped = 'coord0'
+    original_da, expected_da = make_drop_example_dataarray(
+        coord_to_be_dropped=coord_to_be_dropped)
+    assert sc.identical(original_da.drop_coords(coord_to_be_dropped), expected_da)
```

Comment:
The tests become hard to read with this massive function that hides what is going on. Just from looking at these lines it is hard to tell for the reader what is going on. Have you considered being more explicit, something like:

```
da.coords['abc'] = abc
assert 'abc' not in da.drop_coords('abc').coords
```
and possibly accompany that with a separate test that checks that other coords are untouched?

Reply:
Yes, this is much clearer for the reader! :+1:  Whether it should be two separate test is probably a matter of taste/style.